src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMNavigator.java

```diff
@@ -636,9 +651,11 @@ public ListBoxModel doFillCredentialsIdItems(@AncestorInPath SCMSourceOwner cont
                             ? Tasks.getDefaultAuthenticationOf((Queue.Task) context)
                             : ACL.SYSTEM,
                     context,
-                    StandardUsernameCredentials.class,
+                    StandardCredentials.class,
                     URIRequirementBuilder.fromUri(bitbucketServerUrl).build(),
-                    CredentialsMatchers.anyOf(CredentialsMatchers.instanceOf(StandardUsernamePasswordCredentials.class))
+                    CredentialsMatchers.anyOf(
```

Comment:
`AuthenticationTokens.matcher(BitbucketCredentials.class)` should suffice instead of the credentials matchers

NOTE: you will need to use an `AuthenticationTokenContext` to handle the different types of auth for `http` vs `https` urls, so at that point you will change to `AuthenticationTokens.matcher(BitbucketCredentials.class, authContext)`

NOTE2: This comment applies to everywhere you are using `CredentialsMatchers.instanceOf`